src/util/range.h

```diff
@@ -341,6 +341,26 @@ struct ranget final
     return begin_value == end_value;
   }
 
+  /// Drop the first \p count elements.
+  /// If the range has fewer elements, drops everything.
+  /// Similar to \ref skip but modifies the range in-place.
+  /// \return true if the range still contains element after the drop
+  ranget<iteratort> drop(std::size_t count) &&
+  {
+    for(; count > 0 && begin_value != end_value; --count)
```

Comment:
:bulb: Could this be defined in terms of `std::advance` and `std::distance`? I think you could advance the begin iterator by the `std::min` of `count` and the distance between the begin and the end iterators.

Reply:
Maybe, but I don't know if it's worth it. Some distance functions work by iterating from the current position to the end, so using that plus advance would end up being less efficient.